app/experimenter/docs/adrs/0001-create-new-nimbus-application.md

```diff
@@ -0,0 +1,119 @@
+- Cons
+  - Lose single source of truth
+    - Possibly require people to know which application to refer to depending on experiment type
+    - Possibly create an API proxy in front of both to have a single endpoint for dependent callers
+    - More difficult to correlate experiments/learnings across the two applications
+  - Changes that must apply to both Normandy and Nimbus experiments must be made in two places while they're both being actively maintained
+  - Divergence in tooling/processes over time may create confusion when context switching between the two
+  - Possibly still somewhat constricted by past core design decisions
+  - [Python/Django may not align with FXA tools/processes](https://github.com/mozilla/fxa/blob/main/docs/adr/0020-application-architecture.md)
+
+### Start A New Application in TypeScript/Node
+
+Creating a new web application in TypeScript/Node would allow us to completely restart our thinking about the entire structure of the application which would result in both code and workflow/UI that are purpose built from the ground up to solely address the concerns of Nimbus experiments.
+
+- Pros
+  - Totally greenfield approach allows fresh perspectives to flourish
+  - Stronger sense of collective ownership from all new contributors
+  - Every decision and line of code built specifically to address the needs of Nimbus
+  - TypeScript provides excellent feedback and guarantees which improves development experience
+  - Native async/websocket support allows richer UI experience when coupled with SPA frameworks (like React)
+  - [Fits more cohesively into the FXA team's tooling/processes](https://github.com/mozilla/fxa/blob/main/docs/adr/0020-application-architecture.md)
+  - Logic/types in TS can be easily shared across multiple projects potentially including Desktop/Mobile clients
+  - TS may integrate more smoothly with Rust than Python does to more tightly couple the server and the new Experiment Client
+- Cons
+  - Lose single source of truth
+    - Possibly require people to know which application to refer to depending on experiment type
+    - Possibly create an API proxy in front of both to have a single endpoint for dependent callers
+    - More difficult to correlate experiments/learnings across the two applications
+  - Changes that must apply to both Normandy and Nimbus experiments must be made in two places while they're both being actively maintained, but with the added complexity that one is in Python/Django and the other is in TS/Node which increases the context switching complexity
+  - Node application frameworks are less mature than Django
+  - No Django Admin Panel for easy data access
+  - Forced to rewrite all the components that do still apply to both Normandy and Nimbus Experiments
+    - Core db state/logic
+    - Public and Private APIs
+    - Dependent service integration (bugzilla/remote settings/email/auth0)
+    - History/Changelog management
+    - User roles/permissions
```

Comment:
IMHO, we know that engineers tend to have a bias towards clean rewrites, and should mentally weight the complexity of these things-to-be-re-implemented higher in order to try to counter it. Would the re-write still be a good idea if these took twice as long to re-implement as we think they will?

(Conversely, how important are these to the MVP success of Nimbus?)

Reply:
I'm really tempted to refer back to the question of whether Normandy will eventually be deprecated. If so, a rewrite seems much more worthwhile.

Reply:
Yeah that's a good point @rfk, if we restart from scratch I think we'll have to allow for some downtime possibly at the scale of months while the core mechanics of Experimenter that are needed for Nimbus are reimplemented.  It might be possible to reorder the things I listed above to hit a working MVP of a Node implementation for Nimbus sooner, but I think all of those things will be necessary at some point.